stratum/hal/lib/p4/utils.cc

```diff
@@ -113,9 +114,12 @@ std::string P4RuntimeGrpcStatusToString(const ::grpc::Status& status) {
         ::p4::v1::Error detail;
         if (details.details(i).UnpackTo(&detail)) {
           ss << "\n(error #" << i + 1 << ": error code: "
-             << ::google::rpc::Code_Name(ToGoogleRpcCode(detail.code()))
+             << ::google::rpc::Code_Name(
+                    ToGoogleRpcCode(detail.canonical_code()))
```

Comment:
Revisit and make separate PR. `canonical_code` should be the correct field over vendor-specific `code`.